linker/Linker.Steps/MarkStep.cs

```diff
@@ -1701,6 +1785,29 @@ protected virtual void DoAdditionalMethodProcessing (MethodDefinition method)
 		{
 		}
 
+		protected virtual bool ShouldMarkAsInstancePossible (MethodDefinition method)
+		{
+			// We don't need to mark it multiple times
+			if (Annotations.IsInstantiated (method.DeclaringType))
+				return false;
+
+			if (method.IsConstructor && !method.IsStatic)
+				return true;
+
+			if (method.DeclaringType.IsInterface)
+				return true;
+
+			return false;
+		}
+
+		protected virtual void MarkRequirementsForInstantiatedTypes (TypeDefinition type)
+		{
+			Annotations.MarkInstantiated (type);
```

Comment:
Isn't this redundant?

Reply:
In some cases yes.

When `MarkRequirementsForInstantiatedTypes ` is called from `MarkType`

```
			if (type.IsInterface) {
				// There's no benefit to deferring processing of an interface type until we know a type implementing that interface is marked
				MarkRequirementsForInstantiatedTypes (type);
			} else if (type.IsValueType) {
				// Note : Technically interfaces could be removed from value types in some of the same cases as reference types, however, it's harder to know when
				// a value type instance could exist.  You'd have to track initobj and maybe locals types.  Going to punt for now.
				MarkRequirementsForInstantiatedTypes (type);
			} else {
				// There are a number of markings we can defer until later when we know it's possible a reference type could be instantiated
				// For example, if no instance of a type exist, then we don't need to mark the interfaces on that type
				_onlyIfInstantiated.Enqueue (type);
			}
```

`Annotations.MarkInstantiated (type)` is *not* redundant.

When `MarkRequirementsForInstantiatedTypes ` is called from `ProcessMethod`

```
if (ShouldMarkAsInstancePossible (method))
    Annotations.MarkInstantiated (method.DeclaringType);
```

Yes, it *is* redundant.

Now that you point it out.  If I unify this behavoir, I think I can get rid of the entire `_onlyIfInstantiated` queue processing.....I'll try it.

Reply:
To my slight surprise, by changing the usage in `Processmethod` to be 

```
if (ShouldMarkAsInstancePossible (method))
    MarkRequirementsForInstantiatedTypes (method.DeclaringType);
```

I was able to remove the entire `ProcessOnlyIfInstantiated()` mechanism.